rasa/shared/core/domain.py

```diff
@@ -598,9 +600,43 @@ def _collect_overridden_default_intents(
         }
         return sorted(intent_names & set(rasa.shared.core.constants.DEFAULT_INTENTS))
 
-    @staticmethod
+    @classmethod
+    def _update_slots_with_global_not_intent_parameter(
```

Comment:
I think this method should leave the domain in a consistent state. Currently the slot mappings can either be in `required_slots` or at the level above. For the sake of other components relying on this we should in my opinion make sure that the domain always moves the slots mappings beyind `required_slots`.